src/unit_tests/analysisd/test_decoder_syscollector.c

```diff
@@ -1726,153 +3321,154 @@ void test_syscollector_dbsync_hardware_valid_msg_inserted(void **state)
+            "Unable to send packages information to Wazuh DB.");
+
+    ret = DecodeSyscollector(lf, &sock);
+    assert_int_not_equal(ret, -1);
 
+    // Return ok after query error
+    ret = DecodeSyscollector(lf, &sock);
     assert_int_not_equal(ret, 0);
-}
 
-void test_syscollector_dbsync_with_no_type(void **state)
-{
-    Eventinfo *lf = *state;
-    int sock = 1;
-    const char type[] = {"dbsync"};
-    char error_log[128];
-    sprintf(error_log, INVALID_PREFIX, type);
+    // Return ok after query error, changing package_id
+    prev_package_id++;
+    expect_any(__wrap_wdbc_query_ex, *sock);
+    expect_string(__wrap_wdbc_query_ex, query, query);
+    expect_any(__wrap_wdbc_query_ex, len);
+    will_return(__wrap_wdbc_query_ex, result_not_ok);
+    will_return(__wrap_wdbc_query_ex, 0);
+    expect_string(__wrap__mdebug1, formatted_msg, 
+            "Unable to send packages information to Wazuh DB.");
 
-    expect_string(__wrap__merror, formatted_msg, error_log);
-    expect_string(__wrap__mdebug1, formatted_msg, UNABLE_TO_SEND_INFORMATION_TO_WDB);
-    int ret = DecodeSyscollector(lf, &sock);
-    assert_int_equal(ret, 0);
-}
+    ret = DecodeSyscollector(lf, &sock);
+    assert_int_not_equal(ret, -1);
 
-void test_syscollector_dbsync_valid_msg_query_error(void **state)
-{
-    Eventinfo *lf = *state;
+    // Invalid parse out
+    error_package = 0;
+    expect_any(__wrap_wdbc_query_ex, *sock);
+    expect_string(__wrap_wdbc_query_ex, query, query);
+    expect_any(__wrap_wdbc_query_ex, len);
+    will_return(__wrap_wdbc_query_ex, result_not_ok);
+    will_return(__wrap_wdbc_query_ex, 0);
+    expect_string(__wrap__mdebug1, formatted_msg, 
+            "Unable to send packages information to Wazuh DB.");
 
-    const char *query = "agent 001 dbsync hotfixes MODIFIED "
-        "{"
-            "\"hotfix\":\"KB123456\","
-            "\"checksum\":\"abcdef0123456789\""
-        "}";
-    const char *result = "fail";
+    ret = DecodeSyscollector(lf, &sock);
+    assert_int_not_equal(ret, -1);
+}
+
+void test_syscollector_port_valid (void **state)
+{
+    Eventinfo *lf = *state;
+    const char *query = "agent 001 port save 100|2021/10/29 14:26:24|protocol|"
+            "0.0.0.0|10|0.0.0.1|11|12|13|14|ok|15|process";
+    const char *result = "ok ";
     int sock = 1;
 
     expect_any(__wrap_wdbc_query_ex, *sock);
     expect_string(__wrap_wdbc_query_ex, query, query);
     expect_any(__wrap_wdbc_query_ex, len);
     will_return(__wrap_wdbc_query_ex, result);
-    will_return(__wrap_wdbc_query_ex, -1);
-
-    expect_string(__wrap__mdebug1, formatted_msg, UNABLE_TO_SEND_INFORMATION_TO_WDB);
-    expect_string(__wrap__mdebug2, formatted_msg, WDBC_QUERY_EX_ERROR);
+    will_return(__wrap_wdbc_query_ex, 0);
 
     int ret = DecodeSyscollector(lf, &sock);
 
-    assert_int_not_equal(ret, 1);
+    assert_int_not_equal(ret, 0);
 }
 
-void test_syscollector_dbsync_os_valid_msg_no_result_payload(void **state)
+extern int error_port;
```

Comment:
Please, move the global declarations above.

Reply:
In this approach, try to follow the general guideline of declaring variables before their use, which facilitates a better understanding of the code. However, from now on it can be subject to debate.